File: Apis/Domain/Entities/ClassProject.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

public class ClassProject :BaseEntity
{
    public int? Classid { get; set; }


    public int? Projectid { get; set; }

    public virtual Class? Class { get; set; }

    public virtual Project? Project { get; set; }
}

File: EXE201_Heureux/src/main/java/com/example/exe201_heureux/entity/ClassProject.java
package com.example.exe201_heureux.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "Class_Project")
public class ClassProject {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "classid")
    private Class classid;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "projectid")
    private Project projectid;

}